srcs/close.c

```diff
@@ -6,32 +6,63 @@
 /*   By: gudias <marvin@42lausanne.ch>              +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/08/20 17:45:50 by gudias            #+#    #+#             */
-/*   Updated: 2022/08/29 16:58:13 by tgrivel          ###   ########.fr       */
+/*   Updated: 2022/09/05 01:34:04 by gudias           ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "cub3d.h"
 
 static void	destroy_images(t_info *info)
 {
-	mlx_destroy_image(info->mlx[INIT], info->mm_img[GROUND]);
-	mlx_destroy_image(info->mlx[INIT], info->mm_img[WALL]);
-	mlx_destroy_image(info->mlx[INIT], info->mm_img[PLAYER]);
+	if (info->mm_img[GROUND])
+		mlx_destroy_image(info->mlx[INIT], info->mm_img[GROUND]);
+	if (info->mm_img[WALL])
+		mlx_destroy_image(info->mlx[INIT], info->mm_img[WALL]);
+	if (info->mm_img[PLAYER])
+		mlx_destroy_image(info->mlx[INIT], info->mm_img[PLAYER]);
+	if (info->texture.img_north)
+		mlx_destroy_image(info->mlx[INIT], info->texture.img_north);
+	if (info->texture.img_south)
+		mlx_destroy_image(info->mlx[INIT], info->texture.img_south);
+	if (info->texture.img_east)
+		mlx_destroy_image(info->mlx[INIT], info->texture.img_east);
+	if (info->texture.img_west)
+		mlx_destroy_image(info->mlx[INIT], info->texture.img_west);
+	if (info->texture.img_floor)
+		mlx_destroy_image(info->mlx[INIT], info->texture.img_floor);
+	if (info->texture.img_ceil)
+		mlx_destroy_image(info->mlx[INIT], info->texture.img_ceil);
```

Comment:
Do we create a file `utils_mlx.c`
with function like:
```c
my_destroy_image(void *init, void *img)
{
    if (img)
        mlx_destroy_image(init, img);
}

Reply:
Good idea! Let's do it

Reply:
done: 8c121ed0a3121ccb0c18d25d4b5ec7b24d1265b6